Tweeting new releases

elshafeh opened the pull request:
As discussed with Robert rather than tweeting every single commit, we can tweet only new releases. I've found and added a sendtweet.sh (https://github.com/agubelu/bash-send-tweet) and sent a "test" tweet with it and it works. I added a line in release-ftp.sh to tweet the version-name. That however I haven't tested :)

Commit:
tweeting new releases

robertoostenveld commented:
You just spilled the secret twitter key on the internet! Now anyone can tweet as fieldtriptoolbx. The secrets should be in a separate file that is not shared in this repository. 

I will regenerate the secrets.

robertoostenveld commented:
here is some relevant information to review https://help.twitter.com/en/rules-and-policies/twitter-automation

robertoostenveld commented:
> I will regenerate the secrets.

done

robertoostenveld commented on sendtweet:
```diff
@@ -0,0 +1,142 @@
+# You can also define these values in another file named .twitterkeys
+# by placing it in the same directory.
+
+CONSUMER_KEY='SPg4RX8Zdn5nxzP01IgCQGGQB'
```
this should remain in an external file, which (thanks to gitignore) is not added to the public repository.

robertoostenveld commented on sendtweet:
```diff
@@ -0,0 +1,142 @@
+ACCESS_TOKEN_SECRET='Y5VgH8ewwYNjB2GvNgTkEGROnnmne76cwUBYjAGhE4lIU'
+
+if [ -f .twitterkeys ]; then
+    source .twitterkeys
```
this file should be added to .gitignore, not to the repository

robertoostenveld commented on sendtweet:
```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
```
I suggest to add a link to the original source of this script. The license is compatible with ours, so fine to include.

robertoostenveld commented on release-ftp.sh:
```diff
@@ -64,6 +64,9 @@ else
   cp daily/buffer-$TODAY.zip          /home/common/matlab/fieldtrip/data/ftp/modules
 
+  # Tweet about it using (sendweet.sh)
+  ./sendtweet "A New fieldtrip version "$TODAY" is out! Make sure you have our latest and finest :) "
```
Should we include information about the changes? Or a download link? 

What do the @fieldtrip/maintainers think of this?

elshafeh committed:
Update sendtweet

would this then launch the secret.sh?
@robert: have you updated the secret.sh with the new tokens?

robertoostenveld commented:
should we make a short summary of changes available somewhere? I.e. release notes like this?

```
mac036> git log --pretty=tformat:'%h %s' 20191028..20191111
```

(see also https://git-scm.com/docs/pretty-formats) where the sha256 hash could link to the respective commit? And the tweet itself would point to the online location of the release notes. That could for example be <http://www.fieldtriptoolbox.org/release/20191111>

The page content could be something like this

## FieldTrip release 20191111

Please download this version from <ftp://ftp.fieldtriptoolbox.org/pub/fieldtrip/> or <https://github.com/fieldtrip/fieldtrip/releases>

### Changes in this release

- [26cb262ff](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) automatically synchronized identical files to ea3a6e9715250c7df5dc0456337c567e469b215d
- [ea3a6e971](https://github.com/fieldtrip/fieldtrip/commit/ea3a6e971) Update ft_datatype_sens.m
- [8fc36cb46](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) ENH - fixed bug with administration of channel order
- [a3365dd8d](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) ENH - removed transpose, as suggested by Georgios
- [26b68c5b9](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) ENH - fix for reading events.tsv
- [03be9f6ad](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) ENH - major code cleanup for readability, speed up of computations with many observations, allowing for a low-dim sqrtm representation of the kernel
- [08fb2890c](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) updated link to online documentation
- [7dd7a2139](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) automatically synchronized identical files to b40e9f74cd08b237b357a29e42bb335249eed55b
- [b40e9f74c](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) Merge branch 'master' of github.com:fieldtrip/fieldtrip
- [a5a0fb73f](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) automatically synchronized identical files to 896d43b91668165010807622f7407fefd08c3e63
- [03780ac36](https://github.com/fieldtrip/fieldtrip/commit/26cb262ff) FIX - append more than one structure correcly (would lead to error)

robertoostenveld commented on sendtweet:
```diff
@@ -20,13 +22,8 @@
-
-if [ -f .twitterkeys ]; then
-    source .twitterkeys
+if [ -f .secret ]; then
```
it should be `secret.sh`, that is the Bash file which is processed inline. See https://linuxize.com/post/bash-source-command/

elshafeh committed:
Update to sendtweet

added .sh

robertoostenveld commented:
since there is no progress on this, let me close it. The idea of sending an informative tweet upon a release is still interesting, but it should be better than `"A new FieldTrip version "$TODAY" is out! Make sure you have our latest and finest :)`

Feel free to reopen.
